Dockerfile

```diff
@@ -12,11 +12,15 @@ RUN mkdir -p /soils
 COPY soils /soils/soils
 COPY requirements.txt /soils/requirements.txt
 COPY gunicorn.py /soils/gunicorn.py
+COPY gunicorn_dev.py /soils/gunicorn_dev.py
 
 
 WORKDIR /soils
 RUN pip install -r requirements.txt
 
+# Install Flask and Jinja2
+RUN pip install --upgrade flask jinja2
```

Comment:
since there's already a "pip install -r -requirements.txt" above, it's not clear why this is here.